### PR TITLE
feat: support byte string literal in bq

### PR DESCRIPTION
**Pull request by togami2864**
https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#string_and_bytes_literals

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -493,6 +499,25 @@ impl<'a> Tokenizer<'a> {
                     }
                     Ok(Some(Token::Whitespace(Whitespace::Newline)))
                 }
+                // BigQuery uses b or B for byte string literal
```
Comment (AugustoFKL) [inline src/tokenizer.rs:502]: This is something related to BigQuery, as I couldn't find this on MySQL. Maybe we could make this dialect-specific so we avoid permissive syntax on other dialects...?

@alamb

**Review by AugustoFKL — CHANGES_REQUESTED** [review]
Left a comment.

One thing that bothers me is that you didn't test the 'b' literal. Please add some tests for it, if possible.

File: src/tokenizer.rs (hunk 1 above)
Comment (togami2864) [reply src/tokenizer.rs:502]: How about use `dialect_of!` like this.
```rs
b @ 'B' | b @ 'b' if dialect_of!(self is BigQueryDialect | GenericDialect) => { ..
```

**Comment by togami2864** [comment]
We can only choose between 'b' or 'B' when converting AST into string result for test. So, I didn't test it. Is there any good approach??

File: src/tokenizer.rs (hunk 1 above)
Comment (AugustoFKL) [reply src/tokenizer.rs:502]: Seems reasonable

**Comment by AugustoFKL** [comment]
@togami2864 wym?

I'd test it by using 'verified statement' usually... Considering the select, you put as an example:

```sql
SELECT B'abc', B"abc"
```

The resulting query is equals to the input? If so, use the `verified_stmt` method to show that. This way we can assure how the output is based on the input

**Review by alamb — COMMENTED** [review]
This PR is looking good to me - thank you for the work @togami2864  and your help getting this ready for merge @AugustoFKL 

It looks like CI is failing (though that may be resolved with a merge from main to get the clippy fixes). Is there anything else we are waiting on?

**Comment by togami2864** [comment]
All tasks are done. There is nothing else we waiting on.

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @togami2864  and @AugustoFKL

**Pull request merged**